Bottom: 9a878aabd39520f704333e664e41d87a73a56498
Top:    9e4f91cc8f76e52016c05171ea496f1435e8ce6f
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2012-11-20 02:03:38 +0000

autoroute.c: Destroy debug graphics context when we're finished with it

I caught this one by inspection, as we don't tend to build with autorouter
debugging enabled. I tested quickly before comitting, and this change
doesn't break anything. This said, I couldn't see any debug rendering
output either (before or after this patch).. so that might be broken
somewhere.


---

diff --git a/src/autoroute.c b/src/autoroute.c
index a2a64c6..d7d6aa1 100644
--- a/src/autoroute.c
+++ b/src/autoroute.c
@@ -5333,7 +5333,10 @@ donerouting:
     }
 #ifdef ROUTE_DEBUG
   if (ddraw != NULL)
-    ddraw->finish_debug_draw ();
+    {
+      ddraw->graphics->destroy_gc (ar_gc);
+      ddraw->finish_debug_draw ();
+    }
 #endif
 
   if (changed)
